README.md

```diff
@@ -6,109 +6,87 @@
 [![CircleCI](https://circleci.com/gh/freechipsproject/chisel3/tree/master.svg?style=shield)](https://circleci.com/gh/freechipsproject/chisel3/tree/master)
 [![GitHub tag (latest SemVer)](https://img.shields.io/github/tag/freechipsproject/chisel3.svg?label=release)](https://github.com/freechipsproject/chisel3/releases/latest)
 
-Chisel is a new hardware construction language to support advanced hardware design and circuit generation.
+Chisel is a powerful hardware construction language to support advanced hardware design and circuit generation.
 The latest iteration of [Chisel](https://chisel.eecs.berkeley.edu/) is Chisel3,
-which uses Firrtl as an intermediate hardware representation language.
+which uses Firrtl as an intermediate hardware representation language. Chisel promotes re-use through parameterization and 
+the ability to build libraries. Better re-use makes hardware engineers more effective.
+
+# What does Chisel code look like?
+
+Chisel is a [Scala](https://www.scala-lang.org/) embedded DSL, it takes advantage of object oriented and functional techniques to create complex
+circuit generators with a minimum of code. Parameterized generators with strong type and error checking builds re-usable stuff fast.
+```scala
+// This FIR filter has parameterized window length, IO bitwidth, and windowing function
+class MyFir(length: Int, bitwidth: Int, window: (Int, Int) => Seq[Int]) extends Module {
```

Comment:
More importantly, if you want to showcase the power of abstraction, consider starting with how you use the FIR generator, then showing the implementation. Because most users will probably import and call an off-the-shelf FIR generator library.